todo/src/reducer/reducer.js

```diff
@@ -0,0 +1,37 @@
+import { ADD_TODO, TOGGLE_TODO, DELETE_TODO } from '../actions/actions';
+
+
+const initialState = {
+    todos: []
+};
+
+export default function reducer(state = initialState, action) {
+    switch (action.type) {
+        case ADD_TODO:
+        return {
+            ...state,
+            todos: [...state.todos, { value: action.todo, completed: false }]
+          }
+
+        case TOGGLE_TODO:
+            return Object.assign({}, state, {
+                todos: state.todos.map(todo => {
```

Comment:
If the value of two todos is ever duplicate, this breaks! You can solve this by comparing against a unique id of the todo instead of the value.